src/log/context.cr

```diff
@@ -0,0 +1,35 @@
+# Immutable structured context information for logging.
+#
+# See `Log.context`, `Log.context=`, `Log::Context#clear`, `Log::Context#set`, `Log::Context#using`.
+class Log::Context
+  Crystal.datum types: {bool: Bool, i: Int32, i64: Int64, f: Float32, f64: Float64, s: String, time: Time}, hash_key_type: Symbol, immutable: true
+
+  # Creates an empty `Log::Context`
+  def initialize
+    @raw = Hash(Symbol, Context).new
```

Comment:
Why Symbol?

Reply:
Since the context are changed by `Log.context.set key: value` and thowse kwargs have symbols as key it was simple to preserve that type.

Reply:
I guess that means that the context keys can't be dynamic. Is that intended?

Reply:
What I mean is that you can't do something like:

```crystal
h = {:foo => 1, :bar => 2}
Log.context.set h
```

Is that intentional?

Reply:
fixed

Reply:
Cool, now one can pass a hash.

My question was more about why Symbol and not String.

I think we should avoid using Symbol, and eventually remove it from the language. Symbol is a good thing in Ruby, but in Crystal I can't find a really good use case...

Reply:
I still like them. Is a separate domain of simple values for me.

But allowing the backend to filter directly keys based on string operations, and allowing dynamic keys was the selling point here.